### DISCUSSION TITLE
How to rename the sub directory name?

### DISCUSSION THREAD

**Discussion by Dr-AMH** [community/community#31486] [Other Feature Feedback, Questions, & Ideas]
<img width="1274" alt="image" src="https://user-images.githubusercontent.com/77119521/187687388-9b0837a6-5ee7-47d2-a72b-137bc7b22b8b.png">

In the above image I want to change the name wordRiddles_HCG using GUI not CommandLine, so please guide me.

**Comment by jfullstackdev** [comment]
edit any other file inside `wordRiddles_HCG`, then once you are there, look at the directory above, like this image below

![image](https://user-images.githubusercontent.com/47092464/187701589-09975713-9a34-4041-9254-1bbe4bc32271.png)

then hit backspace once the focus is on the filename first letter (before it) being edited, like the image shown above,

then you should be seeing like this 

![image](https://user-images.githubusercontent.com/47092464/187701931-a050c979-8a2b-4745-99b1-bac0285e4364.png)

then change the folder name to your liking

![image](https://user-images.githubusercontent.com/47092464/187702646-e64fed97-ce00-4adb-b17b-1aae7089e669.png)

then between the folder name and file name, type again `/` 

you must be seeing like this

![image](https://user-images.githubusercontent.com/47092464/187702198-b86c7692-e31f-45c0-b4e2-214c50adb926.png)

commit it, and you're done

**Reply by Dr-AMH** [reply]
Thanks for your prompt response.

Actually, in my wordRiddle_HCG, I have only .png files, here you can look.

<img width="1246" alt="image" src="https://user-images.githubusercontent.com/77119521/187705081-763f88b8-e7b2-47f0-a525-817b60066e65.png">


So, now let me know how I can edit? Because what you asked me to do I am trying but nothing happened :(

**Reply by jfullstackdev** [reply]
create at least one file, like `temp.txt`, so you can access the one I presented above, you can delete it later

**Reply by Dr-AMH** [reply]
I did what you said and here the below you can see I have now two folders one wordRiddles_HCG and procedure which you mentioned I followed and have another folder name wordRiddlesImages and only have temp.txt file.

<img width="280" alt="image" src="https://user-images.githubusercontent.com/77119521/187706657-63383b5c-1e4f-4a88-b734-b1bbf1e74f27.png">

**Reply by jfullstackdev** [reply]
the new folder name is `wordRiddlesImages` ? 

if you want to transfer the remaining files, you should be doing it with every file, unfortunately, the remaining files are binary because they are images,

there are two ways you can do it, download the images and reupload it at the new folder,

or move it using GitHub Desktop

I thought in the first place you just want to change the folder name through the UI, which you already achieved

**Reply by Dr-AMH** [reply]
No, the folder name isn't changed, it created a new one and with a new file which is temp.txt.

**Reply by jfullstackdev** [reply]
check this out, 

https://github.blog/2013-03-15-moving-and-renaming-files-on-github/

that's moving files, it's the official documentation and they are calling it as `Moving files to new locations` and when you delete the original one, then that's basically renaming the files through Web UI,

anyways, you can simply create a new directory and upload all of the existing images there, that will achieve the same thing

**Comment by ghost** [comment]
Hi! To solve this problem I created this request here: [Context menu in GitHub](https://github.com/community/community/discussions/31497)

**Reply by Dr-AMH** [reply]
Nice initiative!

**Reply by ghost** [reply]
Hi! thank you for feedback.

**Reply by Dr-AMH** [reply]
actually, this is a very hot issue that how to rename the sub-directory you took this into account and hope we will get a response soon.

**Reply by ghost** [reply]
If GitHub had the context menu feature, you could go into folders and rename easily.

**Reply by Dr-AMH** [reply]
> If GitHub had the context menu feature, you could go into folders and rename easily.

but unfortunately, they don't have it right now.

**Reply by ghost** [reply]
So, to resolve this current issue, I tried to request a technical resource for this. If more people vote for this feature I requested, maybe the GitHub team will consider implementing it. And maybe this can help solve your problem here.

**Reply by ghost** [reply]
Dr-AMH Hi! There is an extension/userscript in chrome that refines the experience of using GitHub, I'm thinking of requesting this issue for this specific extension. I hope to help in some way. I now open a thread: [Context menu in GitHub with refined-github #5935](https://github.com/refined-github/refined-github/issues/5935) as promised in the [refined-github](https://github.com/refined-github/refined-github) extension/userscript to see if it is possible to use this context menu feature to rename files, folders or delete and move files and folders as well.

**Comment by jfullstackdev** [comment]
if you are really into it, it's the nature of Git where GitHub is based, take a look at this one 

![image](https://user-images.githubusercontent.com/47092464/187726851-7b92fe28-ea49-40a5-be99-89e4f6605294.png)

https://stackoverflow.com/questions/11183788/in-a-git-repository-how-to-properly-rename-a-directory

that's why same thing is happening at the Web, it's basically moving files to a new location and delete the old one

it's not just like a Windows OS treating files and folders that way because even if a commit is just deleting something, it must be traced back still because it's a revision control and deletions are critical

**Reply by Dr-AMH** [reply]
@jdevstatic through commandLine it will work and move which again creates the same problem, as @codehangen explained but we are in the technology era and I don't think that it should be very hard to rename the folder name, but why does GitHub restrict this simple feature?

**Reply by ghost** [reply]
Hi Dr-AMH! I got a reply from the extension just now. So, this would not be possible for these reasons:
> "I have a few issues with this:
> - the actions you're describing are probably already available via keyboard shortcut, so a context menu would not be any faster
> - GitHub very much looks like a website rather than an app and generally people don't use the context menu to take actions on websites
> - implementing context menu items in extensions is kinda annoying because the event happens in the background page, which then has to synchronize with the current page. Even just registering them as a "feature" would mean having to write a good chunk of code to handle enabled-disabled features. Refined GitHub runs almost exclusively on the main page
> The last part leads me to say no to this. I'm sure it can be part of its own extension though."

I would like to have this feature like you, but I don't think it's possible with based on the response I received. I think @jdevstatic is right. His tip will work in most cases, although it is not the answer to this request.

**Reply by jfullstackdev** [reply]
> @jdevstatic through commandLine it will work and move which again creates the same problem, as @codehangen explained but we are in the technology era and I don't think that it should be very hard to rename the folder name, but why does GitHub restrict this simple feature?

first, it's not the priority of Git or GitHub because that will break system folder, it's not simply like changing folder name in Windows, if you know, for example, how hard it is to follow the rules of namespace in, say, Java alone, 

in small projects that's not problematic, but when you are dealing with real projects, it's that hard actually, so again, it's not the priority but rather, Git is very particular when it comes to changing folder names, again, it's not simply a folder, it's a namespace in an actual project

second, if you are really into it, there is the GitHub API, you can develop it in such a way that the user will only enter the required parameters and your script will do the rest of folder deletions as if it's in Windows system

originally, GitHub is for code alone, but since they are innovative enough, they added features that can be used outside development, like the GitHub Pages and this new Discussion feature, but still it's based on Git, again, they may implement that, but of all the requests, it's not a priority

**Reply by Dr-AMH** [reply]
@codehangen I will give @jdevstatic a try to his tip.

**Comment by hiarmen** [comment]
Use Git Desktop to rename sub-directories.
